base-permissions/src/commonMain/kotlin/Permission.kt

```diff
@@ -40,20 +40,21 @@ interface BasePermissionsBuilder
 
 /**
  * Closure that takes permission and coroutine contexts and creates [PermissionStateRepo].
- * Each platform registers [RepoFactory] in the register permission helper method.
+ * Each platform registers [PermissionStateRepoBuilder] in the register permission helper method.
  */
-typealias RepoFactory = (permission: Permission, coroutineContext: CoroutineContext) -> PermissionStateRepo<*>
+typealias PermissionStateRepoBuilder = (permission: Permission, coroutineContext: CoroutineContext) -> PermissionStateRepo<*>
```

Comment:
I also thought it can be changed to an interface and passed as an anonymous class. In this case, there are more options for mocking. 
(Not a change request, just an option to consider)

Reply:
As we need a mock we can change this to an interface without breaking the API I think.